test/examples/search/warnings.ts

```diff
@@ -84,27 +97,30 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       await kibanaServer.uiSettings.replace({});
     });
 
-    beforeEach(async () => {
-      // reload the page to clear toasts from previous test
-
-      await PageObjects.common.navigateToApp('searchExamples');
-
-      await comboBox.setCustom('dataViewSelector', dataViewTitle);
-      await comboBox.set('searchMetricField', testRollupField);
-      await PageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+    afterEach(async () => {
+      await PageObjects.common.clearAllToasts();
     });
 
     it('shows shard failure warning notifications by default', async () => {
       await testSubjects.click('searchSourceWithOther');
 
-      // toasts
-      const toasts = await find.allByCssSelector(toastsSelector);
-      expect(toasts.length).to.be(3);
-      const expects = ['2 of 4 shards failed', '2 of 4 shards failed', 'Query result']; // BUG: there are 2 shards failed toast notifications
-      await asyncForEach(toasts, async (t, index) => {
-        expect(await t.getVisibleText()).to.eql(expects[index]);
+      // wait for response - toasts appear before the response is rendered
+      let response: estypes.SearchResponse | undefined;
+      await retry.try(async () => {
+        response = await getTestJson('responseTab', 'responseCodeBlock');
+        expect(response).not.to.eql({});
       });
 
+      // toasts
+      // FIXME: This test should trigger in 2 toast notifications, but could get 3 or 4.
+      // FIXME: Re-test after https://github.com/elastic/kibana/issues/139423
+      // const toasts = await find.allByCssSelector(toastsSelector);
+      // expect(toasts.length).to.be(3); // BUG: should be 2, but could be 3 or 4.
+      // const expects = ['2 of 4 shards failed', '2 of 4 shards failed', 'Query result'];
+      // await asyncForEach(toasts, async (t, index) => {
+      //   expect(await t.getVisibleText()).to.eql(expects[index]);
+      // });
```

Comment:
@ppisljar @Dosant this is the part of the test that was originally flaky. This can probably be uncommented after https://github.com/elastic/kibana/pull/139886

Reply:
👍 let's merge https://github.com/elastic/kibana/pull/139886 and then uncomment and see

Reply:
I'm going to work a bit more on this since https://github.com/elastic/kibana/pull/139886 was just merged.